tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -88,41 +89,109 @@ class AppFilterImpl(
       apps
     }
 
-   // If `any` is specified as logicFilter config, then store the value in variable which is used
-   // later for final computation.
-   val appNameFilteredForDisjunction = if (filterAppName.nonEmpty && appArgs.any()) {
-     appNameFiltered
-   } else {
-     Seq.empty[AppFilterReturnParameters]
-   }
-
     val userNameFiltered = if (userName.nonEmpty) {
       val appNamelogicFiltered = if (appArgs.any()) {
         apps
       } else {
         appNameFiltered
       }
-      appNamelogicFiltered.filter(_.appInfo.exists(_.userName.contains(userName)))
+      appNamelogicFiltered.filter(_.appInfo.appStartInfo.exists(_.userName.contains(userName)))
     } else {
       appNameFiltered
     }
 
+    val configFiltered = if (sparkProperties.nonEmpty) {
+      val userNameLogicFiltered = if (appArgs.any()) {
+        apps
+      } else {
+        userNameFiltered
+      }
+
+      // check if key:value pair is provided in the confNames filter. If those are provided
+      // along with just the key configs, then we have to make sure we consider both variations
+      // i.e key:pair and key configs to filter the event logs.
+      if (sparkProperties.toString.contains(":")) {
+        val configFilteredResult = userNameLogicFiltered.filter { appFilterReturnParameters =>
+          val individualConfigs = sparkProperties.map(str => str.split(":"))
```

Comment:
simialr comment as before here, move this out of the filter and exists check so we only have to do it once.